core/src/test/java/com/netflix/iceberg/util/TestBinPacking.java

```diff
@@ -162,14 +162,30 @@ public void testBinPackingLookBack() {
     // 6. [5, 1]
     Assert.assertEquals("1 bin look-back: should merge ones with fives",
         l(l(5, 1), l(5, 1), l(5, 1)), pack(l(5, 1, 5, 1, 5, 1), 8, 1));
+
+    Assert.assertEquals("2 bin look-back: should merge all ones with five when expensiveTaskFirst is enabled",
+        l(l(5, 1, 1, 1), l(5, 5)), pack(l(5, 1, 5, 1, 5, 1), 10, 2, true));
```

Comment:
This doesn't exercise the largest bin first case because it never requires more than 2 bins.

Reply:
I've removed this test for the PR.